CRM/GoCardless/Page/Webhook.php

```diff
@@ -208,10 +208,22 @@ public function doPaymentsConfirmed($event) {
         ]);
       }
 
-      // Now call completetransaction. Note that the only data this updates in
-      // the contribution record is trxn_id and fee_amount (which we don't
-      // supply).
-      civicrm_api3('Contribution', 'completetransaction', $contribution);
+      // Now call Payment.create. This will call Contribution.completetransaction internally
+      civicrm_api3('Payment', 'create', [
+        'contribution_id' => $contribution['id'],
+        'total_amount' => $contribution['total_amount'],
+        'trxn_date' => $payment->charge_date,
+        'trxn_id' => $payment->id,
+        'is_send_contribution_notification' => 0,
+      ]);
+      // @todo: At this point we want to save the trxn_id and receive_date to the contribution.
```

Comment:
`@fixme` instead? I don't like calling `Contribution.create` again because that call has quite a perfomance hit (which probably doesn't matter via IPN) and we should be looking at a "proper" CiviCRM core fix if appropriate in the future, or changing how the GoCardless extension handles things. But this fixes the problem.